src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyTool.java

```diff
@@ -458,13 +458,16 @@ public SortedProperties loadUnsafeMapping( LicenseMap licenseMap,
     {
         Map<String, MavenProject> snapshots = new HashMap<>();
 
-        //find snapshot dependencies
-        for ( Map.Entry<String, MavenProject> entry : artifactCache.entrySet() )
+        synchronized ( artifactCache )
```

Comment:
This synchronized block prevents the `ConcurrentModificationException` - we can't iterate over its `entrySet` and modify it from other threads at the same time. Other operations like put/remove below are ok to use because we use the `synchronizedSortedMap`.